docker/launch.sh

```diff
@@ -99,6 +103,23 @@ then
     NVIDIA=false
 fi
 
+if [[ ${NVIDIA_VERSION} >2 || ${NVIDIA_VERSION} == 0 ]]; then
+    echo "Nvidia version not recognised, please use either -nv 1 or -nv 2"
+    exit 1
+fi
+
+if [[ ${NVIDIA_VERSION} = 1 || ${NVIDIA_VERSION} = 2 ]]; then
+    if [ ${NVIDIA} = false ]; then
+        echo "Nvidia docker version specified but -g (nvidia graphics) was set to false. To enable GPU support set -g to true. Or to disable GPU support, don't specify -nv"
+        exit 1
+    fi
+fi
+
+if [ -z "${NVIDIA_VERSION}" ];
```

Comment:
@carebare47 I guess it would be better to assign default values before validation checks not after.Or you have some reason to do this way?

Reply:
The thinking behind doing it this way was that if the user enters -g false -nv (1 or 2), the script can notify them that this isn't a valid combination. If these defaults are assigned before validation then we've no way of letting the user know that they're trying to do something invalid